File: tests/fixtures/code/sample-csharp/Services/UserService.cs
using SampleApp.Models;

namespace SampleApp.Services
{
    public class UserService
    {
        private readonly IUserRepository _repository;

        public UserService()
        {
            _repository = new InMemoryUserRepository();
        }

        public User FindById(string id)
        {
            return _repository.FindById(id) ?? new User { Id = id, Name = "unknown" };
        }
    }

    internal class InMemoryUserRepository : IUserRepository
    {
        public User FindById(string id)
        {
            return new User { Id = id, Name = "Ada" };
        }

        public void Save(User user)
        {
            // no-op
        }
    }
}

File: tests/fixtures/code/sample-php/src/UserService.php
<?php

namespace App\Services;

class User
{
    public string $id;
    public string $name;

    public function __construct(string $id, string $name)
    {
        $this->id = $id;
        $this->name = $name;
    }
}

interface UserRepository
{
    public function findById(string $id): ?User;
}

class UserService implements UserRepository
{
    public function findById(string $id): ?User
    {
        return new User($id, 'Ada');
    }

    public function save(User $user): void
    {
        // no-op
    }
}
